### PR TITLE
CLN: refactor `Styler._translate` into composite translate functions

### PR DESCRIPTION
**Pull request by attack68**
This composites `Styler._translate()` into:

```
def _translate():
    ...
    self._translate_header()
    ...
    self._translate_body()
    ...
```

where the `_translate_header()` and `_translate_body()` methods are given some documentation explaining what they are building. The code is not fundamentally changed, but variable names are renamed to add clarity to match the documentation and loop comprehensions replace `for loops` where possible.

Some tests are minimally altered since the generic `_element` method now returns more dict keys for some elements.

### CODE REVIEW

**Comment by jreback** [comment]
is there a `_translate_footer()` (maybe currently unused)?

**Comment by attack68** [comment]
> is there a `_translate_footer()` (maybe currently unused)?

No because the html element <tfoot> is not used in the jinja2 templates. 

It might be useful for *easily* adding styling of a `totals` row for example in the future, albeit can be done currently, its just not that *easy*.

File: pandas/tests/io/formats/style/test_style.py [hunk 1]
```diff
@@ -313,26 +317,47 @@ def test_index_name(self):
...
+                    "type": "th",
+                    "value": "A",
+                    "is_visible": True,
+                    "display_value": "A",
+                },
+                {
+                    "class": "blank col0",
+                    "type": "th",
+                    "value": self.blank_value,
+                    "is_visible": True,
```
Comment (jreback) [inline pandas/tests/io/formats/style/test_style.py:343]: so these are just additional attributes that currently are defaulted?

**Review by jreback — COMMENTED** [review]
looks good a question

File: pandas/tests/io/formats/style/test_style.py (hunk 1 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_style.py:343]: yes the new `_element` function outputs 5 dict keys by default for each cell. 

Previously almost all cells needed these 5 and a few only had 3 keys like column header cells. There is no harm adding them, and actually might make future dev easier, for example a `hide_headers()` method to complement the `hide_index` method.

**Comment by jreback** [comment]
thanks @attack68

**Pull request merged**